src/raftstore/store/peer.rs

```diff
@@ -484,6 +494,34 @@ impl Peer {
         StaleState::Valid
     }
 
+    fn next_lease_expired_time(&self, send_to_quorum_ts: Timespec) -> Timespec {
+        // The valid leader lease should be
+        // "lease = election_timeout - (quorum_commit_ts - send_to_quorum_ts)"
+        // And the expired timestamp for that leader lease is "quorum_commit_ts + lease",
+        // which is "send_to_quorum_ts + election_timeout" in short.
+        send_to_quorum_ts + self.election_timeout
+    }
+
+    fn update_leader_lease(&mut self, ready: &Ready) {
+        // Update leader lease variables when the raft state changes
+        if ready.ss.is_some() {
+            let ss = ready.ss.as_ref().unwrap();
+            match ss.raft_state {
+                StateRole::Leader => {
+                    self.leader_lease_expired_time =
```

Comment:
I don't get it. Even `[t3, t3 + 5s)` is not safe for lease read, how can it cause any problem? Local read doesn't perform until the applied term is equal to the leader term. When applied term is equal to the leader term, `[t3, t3 +5s)` becomes a valid lease, it is certainly safe to perform lease read.